### PR TITLE
Fix image layouts

### PR DESCRIPTION
**Pull request by DethRaid**
Fix validation errors about image layouts by inserting image memory barriers in the appropriate places

I've added code to the shaderpack loading code. The new code examines the render graph and find edges between image usage - edges where one render pass writes to a texture that a previous render pass reads from, or vice versa. I also have special barriers for the swapchain images because they need special love

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -310,6 +309,7 @@ if (CMAKE_BUILD_TYPE STREQUAL "Debug" AND NOT NOVA_IN_SUBMODULE)
     set(NOVA_TEST ON)
 endif()
 
+set(NOVA_TEST ON)
```
Comment (Janrupf) [inline CMakeLists.txt]: Thats not good, add at least a check if nova is in a submodule.

File: src/render_engine/vulkan/vulkan_render_engine.cpp [hunk 2]
```diff
@@ -1874,20 +1893,17 @@ namespace nova {
         }
 
         // Find where we are in the list of passes
-        auto itr = render_passes_by_order.rbegin();
-        while(itr != render_passes_by_order.rend()) {
-            if(*itr == pass.data.name) {
+        int32_t idx = 0;    // Signed so it can be negative so the `for` loop will exit
```
Comment (Janrupf) [inline src/render_engine/vulkan/vulkan_render_engine.cpp]: Doesn't compile with `-Werror` due to signed/unsigned compare

**Review by Janrupf — CHANGES_REQUESTED** [review]

**Comment by Janrupf** [comment]
Very nice, can confirm that all validation erros are gone

File: CMakeLists.txt (hunk 1 above)
Comment (DethRaid) [reply CMakeLists.txt]: We need to talk about this actually. The way you were setting NOVA_TEST doesn't work at all for msvc. Is there an easy way to support msvc or do I have to pass the variable on the command line?

Comment (Janrupf) [reply CMakeLists.txt]: I'm passing the variable to the cmake command line, better said Clion does it for me.

**Comment by cwfitzgerald** [comment]
test this please

**Review by Janrupf — APPROVED** [review]

**Pull request merged**